app/questionnaire/router.py

```diff
@@ -71,36 +71,25 @@ def get_next_location_url(self, location, routing_path):
         Get the first incomplete block in section/survey if trying to access the section/survey end,
         and the section/survey is incomplete or gets the next default location if the above is false.
         """
-        current_block_type = self._schema.get_block(location.block_id)["type"]
         last_block_id = routing_path[-1]
-        last_block_type = self._schema.get_block(last_block_id)["type"]
         hub_enabled = self._schema.is_hub_enabled()
+        has_section_summary = self._schema.is_summary_in_section(location.section_id)
```

Comment:
The problem is we are allowing empty dicts for summaries, and there be processed as false in the next if statement. I kind of thought it would be cleaner to do this

Reply:
Yep I've left a comment on validator re the empty summary defintiions. TLDR - we should enforce a minimum of `"enabled": true` such as we do with the hub

Reply:
Completely valid point. I just went with the story description which said they should be empty